Sources/Networking/HTTPClient/HTTPClient.swift

```diff
@@ -665,8 +666,17 @@ extension HTTPClient {
         return true
     }
 
-    internal func isStatusCodeRetryable(_ statusCode: HTTPStatusCode) -> Bool {
-        return self.retriableStatusCodes.contains(statusCode)
+    internal func isRetryable(_ urlResponse: HTTPURLResponse) -> Bool {
+        let isStatusCodeRetryable = self.retriableStatusCodes.contains(urlResponse.httpStatusCode)
+        let isRetryableString = urlResponse.allHeaderFields[ResponseHeader.isRetryable.rawValue] as? String
```

Comment:
I cast this as a `String` because HTTPURLRequests' initializer only accepts strings as the header value:

```swift
public init?(url: URL, statusCode: Int, httpVersion HTTPVersion: String?, headerFields: [String : String]?)
```

Let me know if there's a better way to handle this!

Reply:
Maybe we could use:

https://developer.apple.com/documentation/foundation/httpurlresponse/3240613-value: